content/blog/tidymodels-2024-q1/index.Rmd

```diff
@@ -0,0 +1,203 @@
+
+```{r aorsf}
+rf_spec <- rand_forest() |>
+  set_engine("aorsf") |>
+  set_mode("censored regression")
+
+rf_fit <- rf_spec |>
+  fit(Surv(time, status) ~ age + sex, data = lung)
+
+lung_5 <- lung[1:5, ]
+predict(rf_fit, new_data = lung_5, type = "time")
+```
+
+Some models allow for predictions based on different values for tuning parameter without having to refit the model. In parsnip, we refer to this as ["the submodel trick"](https://parsnip.tidymodels.org/articles/Submodels.html). Some of those models are regularized models fitted with the [glmnet](https://glmnet.stanford.edu/) engine. In censored, the corresponding `multi_predict()` method has now gained the prediction types `"time"` and `"raw"` in addition to the existing types `"survival"` and `"linear_pred"`. 
+
+Let's fit a regularized Cox model to illustrate. Note how we set the `penalty` to a fixed value of `0.1`.
+
+```{r coxnet-fit}
+cox_fit <- proportional_hazards(penalty = 0.1) |>
+  set_engine("glmnet") |>
+  set_mode("censored regression") |>
+  fit(Surv(time, status) ~ ., data = lung)
+```
+
+Predictions made with `predict()` use that penalty value of 0.1. With `multi_predict()`, we can change that value to something different without having to refit. Conveniently, we can predict for multiple penalty values as well.
+
+```{r coxnet-pred}
+predict(cox_fit, new_data = lung_5, type = "time")
+
+mpred <- multi_predict(cox_fit, new_data = lung_5, type = "time", 
+                       penalty = c(0.01, 0.1)) 
+mpred
+```
+
+The resulting tibble is nested by observation to follow the convention of one row per observation. For each observation, the predictions are stored in a tibble containing the penalty value along with the prediction.
+
+```{r coxnet-pred-detail}
+mpred$.pred[[2]]
+```
+
+You can see that the predicted value from `predict()` matches the predicted value from `multi_predict()` with a penalty of 0.1.
+
+
+## Consistent `augment()` for workflows and parsnip models
+
+If you are interested in exploring predictions in relation to predictors, `augment()` is your extened `predict()` method: it will augment the dataset with its predictions. For classification, it will add hard class predictions as well as class probabilities. For regression, it will add the numeric prediction. If the outcome variable is part of the dataset, it also calculates residuals. This has already been the case for fitted parsnip models, and now also the `augment()` method for workflows calculates residuals. 
+
+```{r augment}
+spec_fit <- fit(linear_reg(), mpg ~ ., mtcars)
+wflow_fit <- workflow(mpg ~ ., linear_reg()) %>% fit(mtcars)
+
+augment(spec_fit, mtcars)
+
+augment(wflow_fit, mtcars)
+```
+
+Both methods also appened on the left-hand side of the data frame, rather than the right-hand side. This means that prediction columns are always visible when printed, even for data frames with many columns.
```

Comment:
```suggestion
Both methods also append on the left-hand side of the data frame, rather than the right-hand side. This means that prediction columns are always visible when printed, even for data frames with many columns.
```